hack/generator/pkg/codegen/pipeline_swagger_augment.go

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package codegen
+
+import (
+	"context"
+
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/astmodel"
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/config"
+	"github.com/pkg/errors"
+	"k8s.io/klog/v2"
+)
+
+/* augmentResourcesWithSwaggerInformation creates a PipelineStage to add status information into the generated resources.
+
+This information is derived from the Azure Swagger specifications. We parse the Swagger specs and look for
+any actions that appear to be ARM resources (have PUT methods with types we can use and appropriate names in the
+action path). Then for each resource, we use the existing JSON AST parser to extract the status type
+(the type-definition part of swagger is the same as JSON Schema).
+
+Next, we walk over all the resources we are currently generating CRDs for and attempt to locate
+a match for the resource in the status information we have parsed. If we locate a match, it is
+added to the Status field of the Resource type, after we have renamed all the status types to
+avoid any conflicts with existing Spec types that have already been defined.
+
+*/
+func augmentResourcesWithSwaggerInformation(idFactory astmodel.IdentifierFactory, config *config.Configuration) PipelineStage {
+	return MakePipelineStage(
+		"augmentWithSwagger",
+		"Add information from Swagger specs for 'status' fields",
+		func(ctx context.Context, types astmodel.Types) (astmodel.Types, error) {
+
+			if config.Status.SchemaRoot == "" {
+				klog.Warningf("No status schema root specified, will not generate status types")
+				return types, nil
+			}
+
+			klog.V(1).Infof("Loading Swagger data from %q", config.Status.SchemaRoot)
+
+			swaggerTypes, err := loadSwaggerData(ctx, idFactory, config)
+			if err != nil {
+				return nil, errors.Wrapf(err, "unable to load Swagger data")
+			}
+
+			klog.V(1).Infof("Loaded Swagger data (%v resources, %v other types)", len(swaggerTypes.resources), len(swaggerTypes.otherTypes))
+
+			statusTypes, err := generateStatusTypes(swaggerTypes)
+			if err != nil {
+				return nil, err
+			}
+
+			// put all types into a new set
+			newTypes := make(astmodel.Types)
+			for _, t := range types {
+				newTypes.Add(t)
+			}
+
+			// build the augmenter we will use
+			augmenter := fuseAugmenters( /* put your augmenters here */ )
```

Comment:
At the moment this is a noöp due to having no augmenters, but I hope you can get the idea 🙂